bouncer.go

```diff
@@ -444,3 +468,51 @@ func crowdsecQuery(bouncer *Bouncer, stringURL string, isPost bool) ([]byte, err
 	}
 	return body, nil
 }
+
+func appsecQuery(bouncer *Bouncer, ip string, httpReq *http.Request) error {
+	routeURL := url.URL{
+		Scheme: bouncer.appsecScheme,
+		Host:   bouncer.appsecHost,
+		Path:   "/",
+	}
+	var req *http.Request
+	if httpReq.Body != nil && httpReq.ContentLength > 0 {
+		bodyBytes, err := io.ReadAll(httpReq.Body)
+		if err != nil {
+			return fmt.Errorf("appsecQuery:GetBody %w", err)
+		}
+		httpReq.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+		req, _ = http.NewRequest(http.MethodPost, routeURL.String(), bytes.NewBuffer(bodyBytes))
+	} else {
+		req, _ = http.NewRequest(http.MethodGet, routeURL.String(), nil)
+	}
+
+	for key, headers := range httpReq.Header {
+		for _, value := range headers {
+			req.Header.Add(key, value)
+		}
+	}
+	req.Header.Add(crowdsecAppsecHeader, bouncer.crowdsecKey)
+	req.Header.Add(crowdsecAppsecIPHeader, ip)
+	req.Header.Add(crowdsecAppsecVerbHeader, httpReq.Method)
+	req.Header.Add(crowdsecAppsecHostHeader, httpReq.Host)
+	req.Header.Add(crowdsecAppsecURIHeader, httpReq.URL.Path)
+
+	res, err := bouncer.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("appsecQuery %w", err)
+	}
+	if res.StatusCode != http.StatusOK {
+		return fmt.Errorf("appsecQuery statusCode:%d", res.StatusCode)
```

Comment:
In case we encounter error like 401 or 500, we could add some more context so the user know best how to debug:  
401: The remediation component is not authenticated. It must use the same API Key that was generated to pull the local API request
500: An error occurred in the application security component. The remediation component must support a WAAP_FAILURE_ACTION parameter to handle this case

Reply:
https://docs.crowdsec.net/docs/next/appsec/protocol#response-code 

hat need to be in the doc, not in the code

Reply:
Added: CrowdsecAppsecFailureBlock var to handle this behavior